increase .flash speed and keep original bg-color of element

sjieg opened the pull request:
PR https://github.com/watir/watir/pull/695 could not be updated anymore, so hereby the new PR.

I've also added specs that check if the bg-color is the same after flashing. Tested in chrome and ff.

Commit:
Updated flash function
- Now flashes faster
- Element will end up with the same color it started
- Add spec that BG color should stay the same after flashing

Commit:
Removed redundant enters

titusfortner reviewed (commented):
This flashes too quickly for me, but this was what the original watir-classic used, so I'm less against it.

What if we set it to 0.1 by default, and implemented: `element.flash :fast` (0.04sec) & `element.flash :slow` (0.2sec)?

jkotests reviewed (approved):
I don't notice a lot of difference between the delay times, so I'm fine with whatever you two agree on.

Commit:
Multiple updates to the flash function

- Added preset values (:slow, :fast, :long, :rainbow)
- Added documentation
- Added possibility to flash with multiple colors
- Reworked the flashes * 2 loop
- Added corresponding unit tests

sjieg commented:
- Updated default `dalay: 0.1`
- Added possibility to allow multiple colors in 1 flash
- Added preset parameter
```
presets = {
          fast: { delay: 0.04 },
          slow: { delay: 0.2 },
          long: { flashes: 5, delay: 0.5 },
          rainbow: { flashes: 5, color: %w(red orange yellow green blue indigo violet) }
      }
return self.flash(presets[preset]) unless presets[preset].nil?
```
- Reworked how the flash loops to allow multiple colors.

I'm pretty sure this is not the best way to add presets, but this is my solution with my limited skills. Feedback is appreciated.
Let me know if the `:rainbow` preset is too much, I thought it was a nice _fun_ addition for TA'ers to talk about.

titusfortner commented:
This implementation looks solid to me. @jkotests can you double take on the new code?

jkotests commented on lib/watir/js_execution.rb:
```diff
@@ -33,29 +33,43 @@ def fire_event(event_name)
-        nextcolor = n.even? ? color : background_color
+      (flashes * color.length).times do |n|
+        modulo = n % color.length
+        nextcolor = color[modulo]
```
While I think this works, it'd be easier to read as:

~~~~~~~~
(color * flashes).each do |nextcolor|
  element_call { execute_js(:backgroundColor, @element, nextcolor) }
  sleep(delay)
end
~~~~~~~~

jkotests commented on lib/watir/js_execution.rb:
```diff
@@ -33,29 +33,43 @@ def fire_event(event_name)
-      element_color = element_call { execute_js(:backgroundColor, @element) }.strip
-      element_color = 'white' if element_color.empty?
+      color = [color] if color.is_a? String
+      color.push(background_color)
```
You can avoid the conditional by doing:

~~~~~~~~
color = Array(color).push(background_color)
~~~~~~~~

jkotests reviewed (changes requested):
Looks good, but I have some suggestions to make it easier to read.

Commit:
Updated according to @jkotests feedback

sjieg commented:
Thanks for the feedback @jkotests! Always happy to learn more efficient notations.
